arch/arm64/boot/dts/rockchip/rk3588s-orangepi-5.dtsi

```diff
@@ -361,7 +361,7 @@
 
 &usbdrd_dwc3_0 {
 	status = "okay";
-	dr_mode = "otg";
+	dr_mode = "host";
 	usb-role-switch;
```

Comment:
`usb-role-switch` only work with `dr_mode = "otg";`.

You can use the following config:

```
&usbdrd_dwc3_0 {
	status = "okay";
	dr_mode = "otg";
	usb-role-switch;
	role-switch-default-mode = "host";
};
```

Reply:
Are there any way to produce it works/not works?

Reply:
Just try to switch mode:

Not sure if the command is correct. My device have some problems now.

```
echo device >/sys/class/usb_role/usbxxx/role
```

What's more, you can configure the gadget and test with a male-to-male cable.

Reply:
You seem right i can't assign otg role

Reply:
I think overlay is fine for who wants to use OTG mode

Reply:
@hzyitc `/sys/class/usb_role/fc000000/role` is not exists. Howeve, /sys/kernel/debug/usb/fc000000.usb/mode is exists and it prints `device` if i check it with cat